### DISCUSSION TITLE
Rye `pin` vs `use`

### DISCUSSION THREAD

**Discussion by ei-grad** [astral-sh/rye#303] [General]
Hey Python community and Armin,

As a dedicated user of Rye, I wanted to discuss a specific command choice.

Currently, Rye uses "pin" to specify a particular Python version (e.g., "rye pin 3.11"). Personally, I find "rye use 3.11" to be more natural and user-friendly. It clearly communicates the action of setting the active Python version.

I'm curious to know your take on this. What do you think about using "use" instead of "pin" for setting the Python version in Rye?

**Comment by mitsuhiko** [comment]
I don't have the strongest of opinions here but here was my thinking of why I ended up with `pin`:

* `rye override set`: this would be what `rustup` does and i constantly forget about how it works because it's so long. Did not like it
* `rye use`: to me sounds like a "temporary" override, one that does not get persisted (maybe goes away when the session closes, or is stored out of tree). It also sets an analogy to `nvm use` which does something else entirely (single session activation only)
* `rye local`: to pick a local override vs `rye global` to pick a global version (not implemented). I felt it to be quite confusing but some systems use this (afaik `phpenv` and `pyenv`)
* `rye pin`: to me expressed that I am "pinning" a specific version of Python. I took this from [volta](https://docs.volta.sh/guide/) which uses `volta pin`.

**Comment by yorunning** [comment]
`poetry use` is because poetry can manage multiple Python environments within a project, and you can switch between these environments. `rye pin` is because rye only binds to one Python version within a project, and the two commands have completely different semantics.
